src/ScottPlot/Plottable/RadialGaugePlot.cs

```diff
@@ -0,0 +1,709 @@
+                theta -= (isBelow ? 1 : -1) * rects[i].Width / 2 * width_to_angle;
+            }
+        }
+
+
+        /// <summary>
+        /// Measure the characters in a string with no more than 32 characters.
+        /// </summary>
+        /// <param name="gfx"></param>
+        /// <param name="font"></param>
+        /// <param name="clientRectangle"></param>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        private List<RectangleF> MeasureCharactersInWord(Graphics gfx, System.Drawing.Font font, RectangleF clientRectangle, string text)
+        {
+            List<RectangleF> result = new();
+
+            using StringFormat string_format = new();
+
+            string_format.Alignment = StringAlignment.Center;
+            string_format.LineAlignment = StringAlignment.Center;
+            string_format.Trimming = StringTrimming.None;
+            string_format.FormatFlags = StringFormatFlags.MeasureTrailingSpaces;
+
+            CharacterRange[] ranges = new CharacterRange[text.Length];
+            for (int i = 0; i < text.Length; i++)
+            {
+                ranges[i] = new CharacterRange(i, 1);
+            }
+            string_format.SetMeasurableCharacterRanges(ranges);
+
+            // Find the character ranges.
+            RectangleF rect = new RectangleF(0, 0, 10000, 100);
+            Region[] regions =
+                gfx.MeasureCharacterRanges(
+                    text, font, clientRectangle,
+                    string_format);
+
+            // Convert the regions into rectangles.
+            foreach (Region region in regions)
+                result.Add(region.GetBounds(gfx));
+
+            return result;
+        }
+
+        /// <summary>
+        /// Measure the characters in the string.
+        /// </summary>
+        /// <param name="gfx"></param>
+        /// <param name="font"></param>
+        /// <param name="clientRectangle"></param>
+        /// <param name="text"></param>
+        /// <returns></returns>
+        private List<RectangleF> MeasureCharacters(Graphics gfx, System.Drawing.Font font, RectangleF clientRectangle, string text)
+        {
+            List<RectangleF> results = new List<RectangleF>();
+
+            // The X location for the next character.
+            float x = 0;
+
+            // Get the character sizes 31 characters at a time.
+            for (int start = 0; start < text.Length; start += 32)
+            {
+                // Get the substring.
+                int len = 32;
+                if (start + len >= text.Length) len = text.Length - start;
+                string substring = text.Substring(start, len);
+
+                // Measure the characters.
+                List<RectangleF> rects =
+                    MeasureCharactersInWord(gfx, font, clientRectangle, substring);
+
+                // Remove lead-in for the first character.
+                if (start == 0) x += rects[0].Left;
```

Comment:
Our autoformatter was ok with this? I'm mildly surprised.

I normally prefer normal if statements, but idk how Scott feels.

Reply:
I typically go for two lines, but one line is permitted by the autoformatter. 

The recommended option is technically 4 lines (2 curly braces) but I'm not too strict about it.

I wrote some statements yesterday that I think read better as one-liners:

```cs
if (X > Width) { X -= X - Width; }
if (Y > Height) { Y -= Y - Height; }
if (X < 0) { X = -X; }
if (Y < 0) { Y = -Y; }
```